components/pages/warning-notes/ReviewWarningNoteForm/ReviewWarningNoteForm.spec.tsx

```diff
@@ -0,0 +1,370 @@
+  contextFlag: 'A',
+});
+
+const mockedChildrensResident = residentFactory.build({
+  contextFlag: 'C',
+});
+
+describe('<ReviewWarningNoteForm />', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('for any resident (adults or childrens)', () => {
+    beforeEach(() => {
+      (PersonView as jest.Mock).mockImplementation(
+        createMockedPersonView(mockedAdultsResident)
+      );
+
+      jest.spyOn(warningNotes, 'useWarningNote').mockImplementation(() => ({
+        data: mockedWarningNote[0],
+        mutate: jest.fn(),
+        revalidate: jest.fn(),
+        isValidating: false,
+      }));
+
+      render(
+        <AuthProvider user={mockedUser}>
+          <ReviewWarningNoteForm personId={100} warningNoteId={10} />
+        </AuthProvider>
+      );
+    });
+
+    it('should show an error message if a review date is not entered', async () => {
+      fireEvent.submit(screen.getByRole('form'));
+
+      await waitFor(() => {
+        screen.getByText('Enter a date review undertaken', {
+          selector: '.govuk-error-message',
+        });
+      });
+    });
+
+    it('should show an error message if the review date entered is in the future', async () => {
+      setDateFieldValue('reviewDate', addYears(new Date(), 2));
+
+      fireEvent.submit(screen.getByRole('form'));
+
+      await waitFor(() => {
+        screen.getByText("Date review undertaken can't be in the future", {
+          selector: '.govuk-error-message',
+        });
+      });
+    });
+
+    it('should show an error message if an invalid review date is entered', async () => {
+      setDateFieldValue('reviewDate', {
+        date: '56',
+        month: '15',
+        year: '10',
+      });
+
+      fireEvent.submit(screen.getByRole('form'));
+
+      await waitFor(() => {
+        screen.getByText('Must be a valid Date', {
+          selector: '.govuk-error-message',
+        });
+      });
+    });
+
+    it('should show an error message if review details are not entered', async () => {
+      fireEvent.submit(screen.getByRole('form'));
+
+      await waitFor(() => {
+        screen.getByText('Enter details of review', {
+          selector: '.govuk-error-message',
+        });
+      });
+    });
+
+    it("should NOT show an error message if a manager's name is not entered", async () => {
+      fireEvent.submit(screen.getByRole('form'));
+
+      await waitFor(() => {
+        expect(
+          screen.queryByText('Enter a manager’s name', {
+            selector: '.govuk-error-message',
+          })
+        ).toBeNull();
+      });
+    });
+
+    it('should NOT show an error message if a manager discussion date is not entered', async () => {
+      fireEvent.submit(screen.getByRole('form'));
+
+      await waitFor(() => {
+        expect(
+          screen.queryByText('Enter a date discussed with manager', {
+            selector: '.govuk-error-message',
+          })
+        ).toBeNull();
+      });
+    });
```

Comment:
Could we move these from the shared describe into the describe block for an adults resident? The children review forms still have the manager's fields as required.